src/theory/datatypes/project_op.h

```diff
@@ -0,0 +1,59 @@
+/*********************                                                        */
+/*! \file project_op.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mudathir Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief a class for MK_BAG operator
+ **/
+
+#include "cvc4_public.h"
+
+#ifndef CVC4__PROJECT_OP_H
+#define CVC4__PROJECT_OP_H
+
+#include <vector>
+#include <ostream>
+
+namespace CVC4 {
+
+class TypeNode;
+
+/**
+ * The class is an operator for kind project used to project elements in a tuple
+ * It stores the indices of projected elements
+ */
+class ProjectOp
+{
+ public:
+  ProjectOp(std::vector<uint32_t> indices);
```

Comment:
```suggestion
  ProjectOp(const std::vector<uint32_t>& indices);
```

Reply:
I am not sure about the best option to pass a vector to a constructor. Clang-tidy complained about const and & and suggested using pass by value and std::move in the implementation. There is some discussion [here](https://stackoverflow.com/questions/34799259/c11-pass-vector-to-constructor)

Reply:
That's interesting, I was unaware of this. @4tXJ7f maybe could comment, my style has favored const & in this case.

Reply:
I agree with @mudathirmahgoub: his solution always copies into the function argument and then moves it within the constructor (which is constant, like 6 pointer assignments). `const &`, on the other hand, tries to avoid the copy for the argument, but then copies within the constructor.
You pay a small price (one additional move), but there is a catch: if you regularly call this method with something else (for example an initializer list), then this will actually save a full copy.